sw/host/opentitanlib/src/crypto/ecdsa.rs

```diff
@@ -146,6 +164,46 @@ impl EcdsaRawSignature {
     pub fn is_empty(&self) -> bool {
         self.r.iter().all(|&v| v == 0) && self.s.iter().all(|&v| v == 0)
     }
+
+    fn size() -> usize {
+        let sig: EcdsaRawSignature = EcdsaRawSignature::default();
+
+        sig.r.len() + sig.s.len()
+    }
+
+    fn from_der(data: &[u8]) -> Result<EcdsaRawSignature> {
+        let sig = Signature::<NistP256>::from_der(data).with_context(|| "Failed to parse DER")?;
+
+        // R and S are integers in big endian format. The size of the numbers is
+        // not fixed, it could be 33 bytes in case the value has the top byte
+        // top bit set, and a leading zero has to be added to keep the value
+        // positive (50% chance), or it could be shorter than 32 bytes (1/256
+        // chance). Need to get around these issues and convert into little
+        // endian of exactly 32 bytes format expected by the firmware.
+        let mut raw_r: Vec<u8> = sig.r().to_bytes().to_vec();
+        let mut raw_s: Vec<u8> = sig.s().to_bytes().to_vec();
+
+        // Convert to little endian format.
+        raw_r.reverse();
+        raw_s.reverse();
+
+        // Fixed size arrays to store the firmware expected representation.
+        let mut r = [0u8; 32];
+        let mut s = [0u8; 32];
+
+        // Copy only what's necessary and ignore the leading zeros in case of
+        // 33 byte values.
+        let copy_size = r.len().min(raw_r.len());
+        r[0..copy_size].copy_from_slice(&raw_r[0..copy_size]);
+
+        let copy_size = s.len().min(raw_s.len());
+        s[0..copy_size].copy_from_slice(&raw_s[0..copy_size]);
```

Comment:
Could we do away with this copying with [`Vec::resize`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize)?